docs/Tutorial_DiscordMERLIN.ipynb

```diff
@@ -0,0 +1,594 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #24.        cand_idx = np.zeros(n, dtype=bool)
</pre><p>Perhaps, we should rename <code>cand_idx</code> to <code>is_cands</code> and then this would be consistent with other naming schemes in STUMPY and it is obvious that it (i.e., the use of the word "is_") is a boolean numpy array:</p><p> </p><pre class="ql-syntax" spellcheck="false">def _select_candidates(subseqs, min_dist):
    n, m = subseqs.shape # n: number of subsequences, #m: length of each subsequence
&nbsp; &nbsp; is_cands = np.zeros(n, dtype=bool)
&nbsp; &nbsp; excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
&nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; r = m * (1.0 - ((min_dist ** 2.0) / (2.0 * m)))&nbsp;


&nbsp; &nbsp; for i in range(0, n):
&nbsp; &nbsp; &nbsp; &nbsp; add_cand_i = True
&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; if i &gt; excl_zone:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; left_non_trivial_cands = np.flatnonzero(is_cands[:i-excl_zone])
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; R = np.matmul(subseqs[left_non_trivial_cands], subseqs[i])
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; mask = np.flatnonzero(R &gt; r)
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; is_cands[left_non_trivial_cands[mask]] = False


&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; if len(mask):
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; add_cand_i = False
&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; if add_cand_i:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; is_cands[i] = True
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; return is_cands
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/505/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Reply:
Yes. better to return the Boolean array and then get the indices if needed. Will be fixed in the next commits.